---
title: InternalError() constructor
slug: Web/JavaScript/Reference/Global_Objects/InternalError/InternalError
tags:
  - Constructor
  - JavaScript
  - Reference
---
<div>{{JSRef}} {{non-standard_header}}</div>

<p>The <strong><code>InternalError()</code> </strong>constructor creates an error that indicates an error that occurred internally in the JavaScript engine. For example: <strong>"InternalError</strong>: too much recursion".</p>

<h2 id="Syntax">Syntax</h2>

<pre class="syntaxbox">new InternalError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</pre>

<h3 id="Parameters">Parameters</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>Optional. Human-readable description of the error</dd>
 <dt><code>fileName</code></dt>
 <dd>Optional. The name of the file containing the code that caused the exception</dd>
 <dt><code>lineNumber</code></dt>
 <dd>Optional. The line number of the code that caused the exception</dd>
</dl>

<h2 id="Examples">Examples</h2>

<h3 id="Creating_a_new_InternalError">Creating a new InternalError</h3>

<pre class="brush: js">new InternalError("Engine failure");
</pre>

<h2 id="Specifications">Specifications</h2>

<p><span class="pl-s">Not part of any standard.</span></p>

<h2 id="Browser_compatibility">Browser compatibility</h2>

<div>
<div>


<p>{{Compat("javascript.builtins.InternalError.InternalError")}}</p>
</div>
</div>

<h2 id="See_also">See also</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
</ul>
